final/src/main/java/ua/epam/spring/hometask/dao/AuditoriumDAO.java

```diff
@@ -21,4 +26,24 @@
      * @return found auditorium or <code>null</code>
      */
     Optional<Auditorium> getByName(String name);
+
+    /**
+     * Get auditorium on particular <code>dateTime</code>
+     *
+     * @param auditoriums auditoriums for event
+     * @param dateTime    Date and time of aired event
+     * @return auditorium on particular <code>dateTime</code>
+     */
+    Auditorium findAuditoriumOnDateTime(NavigableMap<LocalDateTime, Auditorium> auditoriums, LocalDateTime dateTime);
+
+    /**
+     * Counts how many vip seats are there in supplied <code>seats</code>
+     *
+     * @param vipSeats Vip seats to process
+     * @param seats Seats to process
+     * @return number of vip seats in request
+     */
+    default long countVipSeats(Set<Long> vipSeats, Collection<Long> seats) {
+        return seats.stream().filter(vipSeats::contains).count();
+    }
```

Comment:
This logic can be extracted to a private method in BookingService as there is logic related to work with DB.